docs/ossm/ossm2-migration/cluster-wide/README.md

```diff
@@ -0,0 +1,384 @@
+> [!CAUTION]
+> This procedure may cause a traffic disruption for workloads which are restarted at unexpected time.
+
+##### Create your Istio resource
+1. Find a namespace with 2.6 control plane:
+
+    ```sh
+    oc get smcp -A
+    NAMESPACE      NAME                   READY   STATUS            PROFILES      VERSION   AGE
+    istio-system   install-istio-system   6/6     ComponentsReady   ["default"]   2.6.4     115m
+    ```
+1. Prepare the `Istio` resource yaml named `ossm-3.yaml` to be deployed to the same namespace as the 2.6 control plane:
+
+    Here we are not using any `discoverySelectors` so the control plane will have access to all namespaces. In case you want to define `discoverySelectors`, keep in mind that all data plane namespaces you are planning to migrate from 2.6 must be matched.
+
+    Also note that `default` name with `InPlace` update strategy is used which allows usage of the `istio-injection=enabled` label without a need for a `default` `IstioRevisionTag`. In case you want to use different name or `RevisionBased` update strategy, you will have to configure `default` `IstioRevisionTag`. See procedures above.
+
+    ```yaml
+   apiVersion: sailoperator.io/v1alpha1
+   kind: Istio
+   metadata:
+     name: default # the name and the updateStrategy is significant for injection labels
+   spec:
+     updateStrategy:
+       type: InPlace
+     namespace: istio-system # 3.0 and 2.6 control planes must run in the same namespace
+     version: v1.24.1
+   ```
+1. Apply the `Istio` resource yaml:
+
+    > **_NOTE:_** after next step, both 2.6 and 3.0 control planes will try to inject side cars to all pods in namespaces with the `istio-injection=enabled` label and all pods with the `sidecar.istio.io/inject="true"` label after next restart of the workloads. Workloads should be restarted only after the `maistra.io/ignore-namespace: "true"` label is added (see below).
+    ```sh
+    oc apply -f ossm-3.yaml
+    ```
+1. Verify that new `istiod` is using existing root certificate:
+
+    ```sh
+    oc logs deployments/istiod -n istio-system | grep 'Load signing key and cert from existing secret'
+    2024-12-18T08:13:53.788959Z	info	pkica	Load signing key and cert from existing secret istio-system/istio-ca-secret
+    ```
+##### Migrate Workloads
+1. Add `maistra.io/ignore-namespace: "true"` label to the data plane namespace
+
+    The `maistra.io/ignore-namespace: "true"` label will disable sidecar injection for 2.6 proxies in the namespace. This ensures that 2.6 will stop injecting proxies in this namespace and any new proxies will be injected by the 3.0 control plane. Without this, there will be a conflict and the proxy will not start.
+
+      > **_NOTE:_** that once you apply the `maistra.io/ignore-namespace` label, any new pod that gets created in the namespace will be connected to the 3.0 proxy. Workloads will still be able to communicate with each other though regardless of which control plane they are connected to.
```

Comment:
I'm not clear on this... in the earlier procedures, it was said that the `maistra.io/ignore-namespace` label ensured "that 2.6 will stop injecting proxies in this namespace and any new proxies will be injected by the 3.0 control plane", but here it says that it ensures that "any new pod that gets created in the namespace will be connected to the 3.0 proxy", which sounds a bit different. 

The sentence after - "Workloads will still be able to communicate with each other though regardless of which control plane they are connected to.", makes me wonder what the risk with this procedure is?

Reply:
Both are true, 2.6 control plane will not try to inject any new or restarted pods in that namespace. So restarted and any new pods will be injected by 3.0 control plane.

The problem can occur only during the time when you already have new Istio cr created but `maistra.io/ignore-namespace` label is not yet set. In this period, if you restart the workloads there will be injection conflict.

I will try to reword this.